ballerina/tests/parse_bytes_with_projection_options.bal

```diff
@@ -0,0 +1,333 @@
+    test:assertEquals(response.data.products[0].description, "A powerful laptop for all your computing needs.");
+    test:assertEquals(response.data.products[0].specifications?.storage, "512GB SSD");
+    test:assertEquals(response.data.products[0].specifications?.display, "15.6-inch FHD");
+    test:assertEquals(response.data.products[0].specifications?.processor, "Intel Core i7");
+    test:assertEquals(response.data.products[0].specifications?.ram, "16GB DDR4");
+    test:assertEquals(response.data.products[0].specifications?.graphics, "NVIDIA GeForce GTX 1650");
+    test:assertEquals(response.data.products[0].specifications?.camera, ());
+    test:assertEquals(response.data.products[0].specifications?.battery, ());
+    test:assertEquals(response.data.products[0].specifications?.color, ());
+
+    test:assertEquals(response.data.products[1].length(), 5);
+    test:assertEquals(response.data.products[1].id, 2);
+    test:assertEquals(response.data.products[1].name, "Smartphone");
+    test:assertEquals(response.data.products[1].price, 699.99d);
+    test:assertEquals(response.data.products[1].description, ());
+    test:assertEquals(response.data.products[1].specifications?.storage, "256GB");
+    test:assertEquals(response.data.products[1].specifications?.display, "6.5-inch AMOLED");
+    test:assertEquals(response.data.products[1].specifications?.processor, ());
+    test:assertEquals(response.data.products[1].specifications?.ram, ());
+    test:assertEquals(response.data.products[1].specifications?.graphics, ());
+    test:assertEquals(response.data.products[1].specifications?.camera, "Quad-camera setup");
+    test:assertEquals(response.data.products[1].specifications?.battery, "4000mAh");
+    test:assertEquals(response.data.products[1].specifications?.color, ());
+
+    test:assertEquals(response.data.products[2].length(), 6);
+    test:assertEquals(response.data.products[2].id, 3);
+    test:assertEquals(response.data.products[2].name, "Headphones");
+    test:assertEquals(response.data.products[2].brand, "AudioTech");
+    test:assertEquals(response.data.products[2].price, 149.99d);
+    test:assertEquals(response.data.products[2].description, "Immerse yourself in high-quality sound with these headphones.");
+    test:assertEquals(response.data.products[2].specifications?.processor, ());
+    test:assertEquals(response.data.products[2].specifications?.ram, ());
+    test:assertEquals(response.data.products[2].specifications?.graphics, ());
+    test:assertEquals(response.data.products[2].specifications?.camera, ());
+    test:assertEquals(response.data.products[2].specifications?.battery, ());
+    test:assertEquals(response.data.products[2].specifications?.'type, "Over-ear");
+    test:assertEquals(response.data.products[2].specifications?.wireless, true);
+    test:assertEquals(response.data.products[2].specifications?.noise_cancellation, true);
+    test:assertEquals(response.data.products[2].specifications?.color, "Black");
```

Comment:
Is there anyway we can use data-provider for this?

Check this out [Data Providers in Ballerina Testing](https://ballerina.io/learn/test-ballerina-code/define-data-driven-tests/#use-data-providers)

Reply:
Store the expected result in an array and using for loop will reduce the number of lines for this. Data Provider may not be an option here.